packages/gatsby/src/commands/develop.ts

```diff
@@ -296,29 +319,16 @@ async function startServer(program) {
 
   /**
    * Set up the HTTP server and socket.io.
+   * If a SSL cert exists in program, use it with `createServer`.
    **/
-  let server = require(`http`).Server(app)
+  let server = program.ssl
+    ? https.createServer(program.ssl, app)
+    : new http.Server(app)
 
-  // If a SSL cert exists in program, use it with `createServer`.
-  if (program.ssl) {
-    server = require(`https`).createServer(program.ssl, app)
-  }
   websocketManager.init({ server, directory: program.directory })
   const socket = websocketManager.getSocket()
 
-  const listener = server.listen(program.port, program.host, err => {
-    if (err) {
-      if (err.code === `EADDRINUSE`) {
-        // eslint-disable-next-line max-len
-        report.panic(
-          `Unable to start Gatsby on port ${program.port} as there's already a process listening on that port.`
-        )
-        return
-      }
-
-      report.panic(`There was a problem starting the development server`, err)
-    }
-  })
+  const listener = server.listen(program.port, program.host)
```

Comment:
typescript taught me that this callback function is not called with an error. But good news, we check for the port above and change the port to a new number. So this code was old and unused, but ts taught me to delete it!